anta/cli/__main__.py

```diff
@@ -0,0 +1,44 @@
+# Copyright (c) 2023-2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+"""ANTA CLI."""
+
+from __future__ import annotations
+
+import logging
+import sys
+from typing import Callable
+
+from anta import __DEBUG__
+
+# Note: need to separate this file from _main to be able to fail on the import.
+try:
+    from ._main import cli
+
+except ImportError as exc:
+    if exc.name == "click":
+
+        def build_cli(exception: Exception) -> Callable[[], None]:
+            """Build CLI function using the caught exception."""
+
+            def wrap() -> None:
+                """Error message if any CLI dependency is missing."""
+                logging.error(
+                    "The ANTA command line client could not run because the required "
+                    "dependencies were not installed.\nMake sure you've installed "
+                    "everything with: pip install 'anta[cli]'"
+                )
+                if __DEBUG__:
+                    logging.debug("The caught exception was: %s", exception)
```

Comment:
This is no longer being triggered as before:
BEFORE:
```
(.pyavd_cli) ~/git_projects/anta (main ✗) anta nrfu json 
The ANTA command line client could not run because the required dependencies were not installed.
Make sure you've installed everything with: pip install 'anta[cli]'
(.pyavd_cli) ~/git_projects/anta (main ✗) ANTA_DEBUG=true anta nrfu json 
The ANTA command line client could not run because the required dependencies were not installed.
Make sure you've installed everything with: pip install 'anta[cli]'
The caught exception was: cannot import name 'ResultManager' from partially initialized module 'anta.result_manager.models' (most likely due to a circular import) (/home/cbaillar/git_projects/anta/anta/result_manager/models.py)
```
AFTER:
```
(.pyavd_cli) ~/git_projects/anta (chore/python-debugger ✗) ANTA_DEBUG=true anta nrfu json
Traceback (most recent call last):
  File "/home/cbaillar/.pyenv/versions/3.12.3/envs/.anta/bin/anta", line 5, in <module>
    from anta.cli import cli
ImportError: cannot import name 'cli' from 'anta.cli' (/home/cbaillar/git_projects/anta/anta/cli/__init__.py)
```

Before we were going `from ._main import anta, cli`, and now `from ._main import cli` - cli is imported first so maybe that's causing the different behavior.

Reply:
BUT, when running with this new debugger, we get a pretty nice output:
![image](https://github.com/user-attachments/assets/b5795174-33e9-4e16-b21c-44cc7e14e6ac)